src/main.rs

```diff
@@ -99,29 +93,16 @@ fn notmain() -> anyhow::Result<i32> {
             writeln!(new_linker_script, "{}", line)?;
         }
     }
-    // commit file to disk
-    drop(new_linker_script);
-
-    // invoke the linker a second time
-    let mut c2 = Command::new(LINKER);
-    // add the current dir to the linker search path to include all unmodified scripts there
-    // HACK `-L` needs to go after `-flavor gnu`; position is currently hardcoded
-    c2.args(&args[..2])
-        .arg("-L".to_string())
-        .arg(current_dir)
-        .args(&args[2..])
-        // we need to override `_stack_start` to make the stack start below fake RAM
-        .arg(format!("--defsym=_stack_start={}", new_origin))
-        // set working directory to temporary directory containing our new linker script
-        // this makes sure that it takes precedence over the original one
-        .current_dir(tempdir.path());
-    log::trace!("{:?}", c2);
-    let status = c2.status()?;
-    if !status.success() {
-        return Ok(status.code().unwrap_or(EXIT_CODE_FAILURE));
+    new_linker_script.flush()?;
+
+    {
+        let exit_status = linking::link_modified(&args, &current_dir, &tempdir, new_origin)?;
+        if !exit_status.success() {
+            process::exit(exit_status.code().unwrap_or(EXIT_CODE_FAILURE))
```

Comment:
same comment about `exit`. this call could leave a tempdir around. This snippet does:
``` rust
let td = tempfile::tempdir_in(".").unwrap();
dbg!(td.path());
std::process::exit(0);
```